odoo/addons/base/models/ir_fields.py

```diff
@@ -383,6 +383,34 @@ def db_id_for(self, model, field, subfield, value):
                 {'moreinfo': action})
         return id, field_type, warnings
 
+    def _xmlid_to_record_id(self, xmlid, model):
+        """ Return the record id corresponding to the given external id,
+        provided that the record actually exists; otherwise return ``None``.
+        """
+        import_cache = self.env.context.get('import_cache', {})
+        result = import_cache.get(xmlid)
+        if result is not None:
+            if result[0] != model._name:
+                raise ValueError("External ID %s not with model %r" % (xmlid, model._name))
+            return result[1] or None
```

Comment:
Maybe use unpacking to avoid the need for indexing later e.g.

```suggestion
        res_model, res_id = import_cache.get(xmlid) or (None, None)
        if res_model:
            if res_model != model._name:
                raise ValueError("External ID %s not with model %r" % (xmlid, model._name))
            return res_id or None
```
Also is the `or None` really useful here?

Reply:
I avoided unpacking in the beginning because in a former iteration I would store `None` as well in the dict.
Now it makes sense.  Thanks for pointing that out.